src/libs/libjpeg_turbo/lv_libjpeg_turbo.c

```diff
@@ -37,16 +38,27 @@ static lv_result_t decoder_open(lv_image_decoder_t * decoder, lv_image_decoder_d
                                 const lv_image_decoder_args_t * args);
 static void decoder_close(lv_image_decoder_t * decoder, lv_image_decoder_dsc_t * dsc);
 static lv_draw_buf_t * decode_jpeg_file(const char * filename);
-static bool get_jpeg_size(const char * filename, uint32_t * width, uint32_t * height);
+static uint8_t * alloc_file(const char * filename, uint32_t * size);
+static bool get_jpeg_size(uint8_t * data, uint32_t data_size, uint32_t * width, uint32_t * height);
+static bool get_jpeg_direction(uint8_t * data, uint32_t data_size, uint32_t * orientation);
+static void dispose_buffer(uint8_t * cur_pos, uint8_t * buffer, uint32_t line_index, uint32_t width, uint32_t height,
+                           uint32_t stride, uint32_t angle);
 static void error_exit(j_common_ptr cinfo);
 static void jpeg_decoder_cache_free_cb(lv_image_cache_data_t * cached_data, void * user_data);
 /**********************
  *  STATIC VARIABLES
  **********************/
+const int JPEG_EXIF = 0x45786966; /* Exif data structure tag */
+const int JPEG_BIG_ENDIAN_TAG = 0x4d4d;
 
 /**********************
  *      MACROS
  **********************/
+#define TRANS_32_VALUE(big_endian, data) big_endian ? \
```

Comment:
It's a good idea, but the macro LV_BIG_ENDIAN_SYSTEM is not applicable here. The endianness contained in Exif are only used for the current image, it may be inconsistent with the system's endianness.